kernel/src/hil/digest.rs

```diff
@@ -0,0 +1,38 @@
+//! Interface for Digest
+
+use crate::returncode::ReturnCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a> {
+    /// On error `result` will contain device specific error codes
+    fn hash_done(&'a self, result: Result<(), u32>);
+}
+
+pub trait Digest<'a> {
+    /// Set the client instance which will receive `hash_done()` callbacks
+    fn set_client(&'a self, client: &'a dyn Client<'a>);
+
+    /// Set the hash key
+    fn set_key(&self, key: &[u8], len: usize) -> ReturnCode;
```

Comment:
Technically a digest (cryptographic hash) does not have a key: HMAC using a digest does, however. So I'd suggest pulling `set_key` out of this trait and instead having `set_mode_hmacsha256` take the key. This would also allow you to statically check the key length by specifying it as a reference to an array.

Reply:
Good idea, fixed in the next version.